e2e/components/stepper-e2e.spec.ts

```diff
@@ -73,7 +73,7 @@ describe('stepper', () => {
       nextButton.get(0).click();
 
       expect(await element(by.css('mat-step-header[aria-selected="true"]')).getText())
-          .toBe('1\nFill out your name');
+          .toBe('create\nFill out your name');
```

Comment:
@mmalerba 

- The `completed` and `editable` inputs have no breaking change, they will still work as before. (I need to add `completed` back as an Input).

- `hasError` would create a breaking change; therefore, I will be adding a `showError` input which will allow devs to have the option to display the error or not and by default it will be set to `false`, instead of doing it automatically as I am now. So this would be an enhancement.

- The changes in functionality are now in regards to the done and edit icons. This is because before it would _always_ show the edit icon, regardless of the step being completed. And now, instead of showing the edit icon when it's not the current step, it's **only** showing the edit icon when it _is_ the current step (which is why the tests were changed). If you think that we should also keep the way it was working before then I could add some type of Input which would allow the developer to either use the new functionality or the old one (old one would be set to the default).

Thoughts?

Reply:
Rather than add `@Input()`s for these, another option is to create an injectable `MAT_STEPPER_GLOBAL_OPTIONS`. We have this for a few other components (e.g. mat-label: https://github.com/angular/material2/blob/d6fec3573cbc854d5bf67d920c5b0f8a6eef0c86/src/lib/core/label/label-options.ts). This way people could enable it for all steppers in their application simultaneously

Reply:
Ok I am going to try that, thanks!